File: DTC.Api/Dtos/Tournament/UpdateTournamentDto.cs
using DTC.Api.Enums;
using System.ComponentModel.DataAnnotations;

namespace DTC.Api.Dtos.Tournament
{
    public class UpdateTournamentDto
    {
        [Required(ErrorMessage = "Der Name ist ein Pflichtfeld.")]
        [StringLength(100, ErrorMessage = "Der Name darf maximal 100 Zeichen lang sein.")]
        public string Name { get; set; } = string.Empty;

        [StringLength(1000, ErrorMessage = "Die Beschreibung darf maximal 1000 Zeichen lang sein.")]
        public string? Description { get; set; }

        public TournamentStatus Status { get; set; } = TournamentStatus.Draft;

        [Required(ErrorMessage = "Das Startdatum ist ein Pflichtfeld.")]
        public DateTimeOffset StartDate { get; set; }
    }
}

File: dtc.frontend/src/dtos/tournament/UpdateTournamentDto.ts
import z from "zod";
import { TournamentStatus } from "../../enums/TournamentStatus";

export const updateTournamentDtoSchema = z.object({
  name: z.string().min(1).max(100),
  description: z.string().max(1000).nullable(),
  status: z.enum(TournamentStatus),
  startDate: z.string(),
});

export type UpdateTournamentDto = z.infer<typeof updateTournamentDtoSchema>;
